xml/docker_building_images.xml

```diff
@@ -150,6 +150,25 @@
    </para>
   </note>
 
+  <note>
+   <title>Building images on systems registered with &rmt;</title>
+
+   <para>
+    When the host system used for building the docker image is registered
+    against &rmt; it is only possible to build containers for the same &slea; code
+    base that the host system is running on. I.e. if you docker host is a
+    &slea;&nbsp; 15 system you can only build &slea;&nbsp; 15 based images
+    on that host by default.
```

Comment:
Generally speaking, that shouldn't be the case.

Reply:
@ikapelyukhin What exactly shouldn't be the case?

You mean it should be possible to build SLE12 containers on a SLE15 host when that host is registered against RMT? If yes, how so. AFAICS the RMT APIs are not providing the information to make that possible.

Reply:
@rhafer well, what information do you need? :slightly_smiling_face:

Reply:
@ikapelyukhin In order to fetch the repos for a given product `container-suseconnect` uses the `/connect/subscriptions/products` API for that to work it needs the regcode. The tool currently queries that via `/connect/systems/subscriptions` (using the credentials from the SCCcredentials file on the host). This all works fine when connecting to SCC directly.

However when connecting to RMT (or SMT) the `/connect/systems/subscriptions` API doesn't exist and  `/connect/subscriptions/products` will give a 401 (unauthorized) as it needs a regcode. In that case `container-suseconnect` will fallback (at least when using https://github.com/SUSE/container-suseconnect/pull/38)  to `/connect/systems/products` which just returns the repos for the product(s) that the host's credentials have been registered for.

So if you know a way to workaround that I am happy to learn about that.

Reply:
@rhafer you can do the following experiment: see in how many places in SUSEConnect's code base `subscriptions/products` is being accessed.

It isn't used by SUSEConnect at all. I believe `subscriptions/products` endpoint is only used by SUMA to do subscription matching. You should never use `subscriptions/products` (probably shouldn't use `systems/subscriptions` either) -- always use `systems/products` instead.

On public cloud RMTs you are likely to run into other authentication issues, but outside of the public cloud RMT's authentication mechanisms are much less restrictive than that of SCC.

Reply:
> It isn't used by SUSEConnect at all. I believe `subscriptions/products` endpoint is only used by SUMA to do subscription matching. You should never use `subscriptions/products` (probably shouldn't use `systems/subscriptions` either) -- always use `systems/products` instead.

@ikapelyukhin Note, that the original code in container-suseconnect isn't from me, but I am pretty sure that it's implement the way it is intentionally.  E.g. to specifically allow building e.g. SLE12 images on SLE15 hosts.

Moving to `systems/products` in all cases would "break" what I think is quite a handy feature. Even when using SCC.

What's wrong with the current approach in https://github.com/SUSE/container-suseconnect/pull/38? It actually works for SCC and RMT (with the restriction mention in here)

Maybe @flavio can shed some light on this as he wrote much of the original code I believe.

(Somehow I think this discussion should have move to https://github.com/SUSE/container-suseconnect/pull/38 a couple of comments ago, but it's to late for that now, ;-) )

Reply:
@rhafer ah, OK -- `container-suseconnect` implements a zypper plugin and `systems/subscriptions` is used to get repo access tokens from SCC. Fair enough, this works for now. I've sent out an email with further details about this issue in general, see if for further details about the build process in public clouds.